my_app/lib/main.dart

```diff
@@ -74,13 +80,12 @@ class _MyHomePageState extends State<MyHomePage> {
             alignment: Alignment.bottomRight,
             child: Padding(
               padding: const EdgeInsets.only(bottom: 20, right: 42),
-              child: BlocBuilder<CalculatorCubit, CalculatorState>(
-                builder: (context, state) {
-                  return Text(state.showDecimal ? state.displayNum.toString() : state.displayNum.toInt().toString(),
-                    style: Theme.of(context).textTheme.bodyLarge,
-                  );
-                }
-              )
+              child: Text(
```

Comment:
context.select will trigger the build function and rebuild everything inside. Consider using consumers (https://pub.dev/documentation/provider/latest/provider/Consumer-class.html) to rebuild as little widgets as possible.

Reply:
This is useful for small widgets, for bigger widgets you might want to consider extracting it and creating another widget class.

Reply:
Instead of consumer you can use a `Selector` https://pub.dev/documentation/provider/latest/provider/Selector-class.html
(I just discovered it)